creme/linear_model/glm.py

```diff
@@ -62,21 +60,16 @@ def _fit(self, x, y, w, get_grad):
     # Single instance methods
 
     def _raw_dot_one(self, x: dict) -> float:
-        return utils.math.dot(self.weights, x) + self.intercept
+        return self.weights @ x + self.intercept
 
     def _eval_gradient_one(self, x: dict, y: float, w: float) -> (dict, float):
 
+        x = utils.VectorDict(x)
```

Comment:
Not awesome but just to show the limits of the current implementation.
The conversion is not needed for `_raw_dot_one(x)` but it is for `x * loss_gradient`.
Conversions induce a copy, same as any addition or multiplication that is not in-place.
On the other hand the dict comprehension does not make any copy except the creation of the final output.

Reply:
An alternative would be to make conversions and operations lazy, like a computation graph, but it would be quite a bit more complex. For example `(loss_gradient * VectorDict(x) + 2. * self.l2 * self.weights).eval()`.

Reply:
I guess that we can do the conversion for the moment. The lazy computation graph is interesting, and could be considered as a separate project.